services/gcp/cloudsql/instances.yml

```diff
@@ -0,0 +1,163 @@
+      produce a DOS if executed repeatedly.
+  get:
+    risks: [discovery:infra, discovery:policy]
+    notes: >-
+      Allows access to instance metadata and settings, including IP addresses, authorized networks, and backup settings.
+  getDiskShrinkConfig:
+    risks: []
+    notes: >-
+      The specific use of this permission is unknown.
+  import:
+    risks: [impact:consumption, impact:dos, impact:manipulation]
+    notes: >-
+      Allows importing data to an instance. If enough data is imported to exceed the disk space of the instance, 
+      will cause a DOS until the instance is manually resized.
+  list:
+    risks: [discovery:infra, discovery:policy]
+    notes: >-
+      Allows access to instance metadata and settings, including IP addresses, authorized networks, and backup settings.
+  listEffectiveTags:
+    risks: [discovery:policy]
+  listServerCas:
+    risks: []
+  listTagBindings:
+    risks: [discovery:policy]
+  login:
+    risks: []
+    notes: >-
+      The specific use of this permission is unknown.
+  migrate:
+    risks: []
+    notes: >-
+      The specific use of this permission is unknown.
+  performDiskShrink:
+    risks: []
+    notes: >-
+      The specific use of this permission is unknown.
+  promoteReplica:
+    risks: [impact:spend]
+    notes: >-
+      This permission allows converting a read replica to a primary instance.
+    links:
+      - https://cloud.google.com/sql/docs/mysql/replication
+  reencrypt:
+    risks: []
+    notes: >-
+      This permission re-encrypts the instance with the existing primary key.
+  resetReplicaSize:
+    risks: []
+    notes: >-
+      The specific use of this permission is unknown.
+  resetSslConfig:
+    risks: [impact:dos]
+    notes: >-
+      Deletes all client certificates and generates a new server SSL certificate for the instance.
+      This can cause a denial of service since clients will not be updated to use the new certificate.
+    links:
+      - https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances/resetSslConfig
+  restart:
+    risks: [impact:dos]
+    notes: >-
+      Restarts the instance. This can cause a denial of service since it closes all existing connections.
+  restoreBackup:
+    risks: [destruction:data, destruction:logs]
+    notes: >-
+      The restore process overwrites all the current data on the instance and it cannot be recovered.
+    links:
+      - https://cloud.google.com/sql/docs/mysql/backup-recovery/restoring
+  rotateServerCa:
+    risks: [impact:dos]
+    notes: >-
+      Rotates the server certificate to one signed by the certificate authority version previously added. 
+      Can cause a denial of service if clients have not been updated to use the new certificate.
+      There must be another certifcate authority already added to exploit this.
+    links:
+      - https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances/addServerCa
+  startReplica:
+    risks: [impact:spend]
```

Comment:
Does this actually accrue spend? How is the replica billed? Naively I'd expect that the replica is pre-sized, so executing replication shouldn't alter spend.

Reply:
You're right, Google says there's no charge for the replication process, I just missed it. Removed.